arbnode/resourcemanager/resource_management.go

```diff
@@ -33,7 +34,15 @@ var (
 func Init(conf *Config) {
 	if conf.MemoryLimitPercent > 0 {
 		node.WrapHTTPHandler = func(srv http.Handler) (http.Handler, error) {
-			return newHttpServer(srv, newLimitChecker(conf)), nil
+			var c limitChecker
+			var err error
+			c, err = newCgroupsMemoryLimitCheckerIfSupported(conf)
```

Comment:
nit: any reason why not just c,err := ... ; and drop declaration of those variables (two lines above this one) ?

Reply:
I wanted to have c be of the `limitChecker` interface type so I can assign the `trivialLimitChecker` to it. I moved the declaration `err` into the same line as the limit checker creation.